File: C/0017.c
#include <stdio.h>
#include <string.h>

// Function prototypes
void number_to_words(int num, char *words);
int count_letters(const char *str);

// Arrays for word representations
const char *units[] = {"", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
const char *teens[] = {"", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"};
const char *tens[] = {"", "ten", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"};
const char *thousands[] = {"", "thousand"};
const char *hundreds[] = {"", "one hundred", "two hundred", "three hundred", "four hundred", "five hundred", "six hundred", "seven hundred", "eight hundred", "nine hundred"};

// Function to convert a number into words
void number_to_words(int num, char *words) {
    if (num == 1000) {
        strcpy(words, "one thousand");
        return;
    }

    words[0] = '\0'; // Initialize the string

    if (num >= 100) {
        int hundred = num / 100;
        strcat(words, hundreds[hundred]);
        num %= 100;
        if (num > 0) {
            strcat(words, " and ");
        }
    }

    if (num >= 20) {
        int ten = num / 10;
        strcat(words, tens[ten]);
        num %= 10;
        if (num > 0) {
            strcat(words, "-");
        }
    } else if (num >= 11) {
        strcat(words, teens[num - 10]);
        num = 0;
    } else if (num == 10) {
        strcat(words, tens[1]);
        num = 0;
    }

    if (num > 0) {
        strcat(words, units[num]);
    }
}

// Function to count letters in a string, excluding spaces and hyphens
int count_letters(const char *str) {
    int count = 0;
    while (*str) {
        if (*str != ' ' && *str != '-') {
            count++;
        }
        str++;
    }
    return count;
}

int main() {
    int total_letters = 0;
    char words[100];

    for (int i = 1; i <= 1000; i++) {
        number_to_words(i, words);
        total_letters += count_letters(words);
    }

    printf("Total letters used from 1 to 1000: %d\n", total_letters);

    return 0;
}

File: TypeScript/0017.ts
// Function to get the number of letters in the word representation of a number
function countLetters(n: number): number {
    // Predefined mappings for numbers and their letter counts
    const ones = ["", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
    const teens = [
        "ten",
        "eleven",
        "twelve",
        "thirteen",
        "fourteen",
        "fifteen",
        "sixteen",
        "seventeen",
        "eighteen",
        "nineteen",
    ];
    const tens = ["", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"];

    let letters = 0;

    if (n === 1000) {
        // Special case for "one thousand"
        letters += "onethousand".length;
    } else {
        const hundredsDigit = Math.floor(n / 100); // Hundreds place
        const remainder = n % 100; // Remaining two digits

        if (hundredsDigit > 0) {
            // Add letters for "X hundred"
            letters += ones[hundredsDigit].length + "hundred".length;

            if (remainder > 0) {
                // Add "and" if there's a remainder
                letters += "and".length;
            }
        }

        if (remainder > 0) {
            if (remainder < 10) {
                // Single-digit number
                letters += ones[remainder].length;
            } else if (remainder < 20) {
                // Teen numbers
                letters += teens[remainder - 10].length;
            } else {
                // Two-digit number (e.g., "twenty-one")
                const tensDigit = Math.floor(remainder / 10);
                const onesDigit = remainder % 10;
                letters += tens[tensDigit].length + ones[onesDigit].length;
            }
        }
    }

    return letters;
}

// Main function to calculate the total number of letters from 1 to 1000

let totalLetters = 0;

for (let i = 1; i <= 1000; i++) {
    totalLetters += countLetters(i);
}

console.log(`The total number of letters used is: ${totalLetters}`);


